src/main/java/tds/exam/results/configuration/web/security/OAuthSecurityConfiguration.java

```diff
@@ -0,0 +1,64 @@
+package tds.exam.results.configuration.web.security;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.client.ClientHttpRequestFactory;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.security.oauth2.client.OAuth2ClientContext;
+import org.springframework.security.oauth2.client.OAuth2RestOperations;
+import org.springframework.security.oauth2.client.OAuth2RestTemplate;
+import org.springframework.security.oauth2.client.resource.OAuth2ProtectedResourceDetails;
+import org.springframework.security.oauth2.client.token.grant.password.ResourceOwnerPasswordResourceDetails;
+import org.springframework.security.oauth2.common.AuthenticationScheme;
+import org.springframework.security.oauth2.config.annotation.web.configuration.EnableOAuth2Client;
+
+@EnableOAuth2Client
+@Configuration
+public class OAuthSecurityConfiguration {
+    private static final String GRANT_TYPE_PASSWORD = "password";
+
+    @Value("${oauth.access.url")
+    private String accessUrl;
+
+    @Value("${oauth.tis.client.id}")
+    private String clientId;
+
+    @Value("${oauth.tis.client.secret}")
+    private String clientSecret;
+
+    @Value("${oauth.tis.username}")
+    private String username;
+
+    @Value("${oauth.tis.password}")
+    private String password;
```

Comment:
Recommend shoving the above properties into a ConfigurationProperties class, and auto-wiring the properties class in a constructor. (And annotate this class with a
`@EnableConfigurationProperties({MyOAuthConfigurationProperties.class})` 
to get it loaded.)

The general rule-of-thumb in Spring 4+ is that @Value is bad.